inc/class-previews.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * Replicating core WP previews for Irving.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving;
+
+/**
+ * Singleton class for customizing core's preview logic as needed to work with
+ * Irving.
+ */
+class Previews {
+
+	/**
+	 * Class instance.
+	 *
+	 * @var null|self
+	 */
+	protected static $instance;
+
+	/**
+	 * Get class instance
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! isset( static::$instance ) ) {
+			static::$instance = new static();
+			static::$instance->setup();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Setup the singleton. Validate JWT is installed, and setup hooks.
+	 */
+	public function setup() {
+
+		// Allow-list some query vars.
+		add_filter( 'query_vars', [ $this, 'modify_query_vars' ] );
+
+		// Filter to easily enable public previews.
+		if ( apply_filters( 'wp_irving_enable_public_previews', '__return_false' ) ) {
+			$this->enable_public_previews();
+		}
+
+		// Validate that JWT exists and is enabled.
+		if ( ! defined( 'JWT_AUTH_VERSION' ) ) {
```

Comment:
This check happens too early, at least in my experience. Perhaps move the creation of the class instance to the `plugins_loaded` hook like you've done in the JWT_Auth class.

Reply:
Good call. Moved into the action.